DCMParser.java

```diff
@@ -0,0 +1,134 @@
+package dcmparser;
+
+import java.io.File;
+import org.dcm4che3.data.Attributes;
+import org.dcm4che3.data.Tag;
+import org.dcm4che3.io.DicomInputStream;
+import org.postgresql.util.PSQLException;
+
+import daoclasses.ImageDAO;
+import daoclasses.PatientStudyDAO;
+import daoclasses.SeriesDAO;
+import ui.Home;
+
+
+
+public class DCMParser {
+	
+	static DicomInputStream din;
+	// dicom file parser
+	public static void dcmFileParser(File f) throws Exception  
+	{		
+		
+		try
+		{   
+			din = new DicomInputStream(f);			
+			Attributes tagDataSet = din.readDataset(-1, -1);	
+			String sdate = tagDataSet.getString(Tag.StudyDate);
+			String stime = tagDataSet.getString(Tag.StudyTime);
+			
+			StringBuffer studyDate = splitDate(sdate);			
+			StringBuffer studyTime = splitTime(stime);
+			//System.out.println(tagDataSet.getString());
+			// creating new patient, study, image obj
+			PatientStudy  ps = new PatientStudy(tagDataSet.getString(Tag.PatientID), tagDataSet.getString(Tag.PatientName), tagDataSet.getString(Tag.PatientBirthDate), tagDataSet.getString(Tag.AccessionNumber), tagDataSet.getString(Tag.StudyInstanceUID), tagDataSet.getString(Tag.StudyDescription), String.valueOf(studyDate.append("  " + studyTime)));
+			Series s = 	new Series(tagDataSet.getString(Tag.SeriesInstanceUID), tagDataSet.getString(Tag.StudyInstanceUID), tagDataSet.getString(Tag.SeriesNumber), tagDataSet.getString(Tag.Modality), tagDataSet.getString(Tag.SeriesDescription));
+			Image i = new Image(tagDataSet.getString(Tag.ReferenceImageNumber), tagDataSet.getString(Tag.StudyInstanceUID), tagDataSet.getString(Tag.SeriesInstanceUID), tagDataSet.getString(Tag.ImageType), tagDataSet.getString(Tag.Rows), tagDataSet.getString(Tag.Columns) , tagDataSet.getString(Tag.BitsAllocated), tagDataSet.getString(Tag.BitsStored));
+			//	Adding to db				
+			
+			PatientStudyDAO.insert(ps);
+			SeriesDAO.insert(s);
+			ImageDAO.insert(i);
+			// 	Displaying info	
+			Home.dispPatient();
+		}
+		catch (PSQLException  e) {
+			throw e;
+		}
+		catch(NullPointerException e)

```

Comment:
You can't catch null pointer, if any doubt ask me

Reply:
While importing a file, if we press cancel without selecting a file it will throws NullPointerException at that time we need to say some message to the user, so that we catch. We'll rectify that.

Reply:
You might need to handle that case.
Like a null check or something and like you said showing a message.

Anyway good reply.